packages/environment-check/src/checks/destination.ts

```diff
@@ -0,0 +1,248 @@
+        const catalog = provider.catalog(odataVersion);
+        result = await catalog.listServices();
+        if (result.length > 0) {
+            const numberOfServices = countNumberOfServices(result);
+            logger.log(
+                t('info.numServicesForDestination', {
+                    odataVersion,
+                    destination: destination.name,
+                    numServicesForDest: getServiceCountText(numberOfServices)
+                })
+            );
+        }
+    } catch (error) {
+        responseStatus = error?.response?.status || error?.cause?.status;
+        logger.error(
+            catalogMessages[responseStatus]
+                ? catalogMessages[responseStatus](destination, odataVersion)
+                : t('error.queryFailure', { odataVersion, destination: destination.name })
+        );
+        const errorJson = error.toJSON ? error.toJSON() : {};
+        if (errorJson?.config?.auth?.password) {
+            delete errorJson.config.auth.password;
+        }
+        logger.info(t('error.urlRequestFailure', { url, error: error.message, errorObj: error }));
+    }
+    return {
+        messages: logger.getMessages(),
+        result,
+        responseStatus
+    };
+}
+
+/**
+ * Returns whether a given destination requires username/password.
+ *
+ * @param destination - the destination to check
+ * @returns boolean if basic auth is required
+ */
+export function needsUsernamePassword(destination: Destination): boolean {
+    return !!destination && destination.credentials?.authentication === 'NoAuthentication';
+}
+
+/**
+ * Checks the destinations and returns a list. Optionally, deep dive into a list of passed destinations.
+ *
+ * @returns messages, destinations
+ */
+export async function checkBASDestinations(): Promise<{
+    messages: ResultMessage[];
+    destinations: Destination[];
+}> {
+    const logger = getLogger();
+    const destinations: Destination[] = [];
+    let url: string;
+
+    // Reload request
+    try {
+        await axios.get(getAppStudioProxyURL() + '/reload');
+    } catch (error) {
+        logger.warning(t('warning.reloadFailure'));
+        logger.info(
+            t('info.urlRequestFailure', {
+                url: `${getAppStudioProxyURL() + '/reload'}`,
+                error: error.message,
+                errorObj: error.toJSON ? JSON.stringify(error.toJSON(), null, 4) : error
+            })
+        );
+    }
+    // Destinations request
+    try {
+        let retrievedDestinations: Destination[] = [];
+
+        if (isAppStudio()) {
+            const response = await destinationsApi.getDestinations();
+            retrievedDestinations = response;
+        } else {
+            // Destination check for VSCode would go here
+        }
+
+        for (const destination of retrievedDestinations) {
+            destination.urlServiceType = getUrlServiceTypeForDest(destination);
+            destination.basProperties.webIDEEnabled = 'true';
```

Comment:
What is the reason for setting this property to true for all destinations? `WebIDEEnabled` already is an optional property on a destination object (https://github.com/SAP/open-ux-tools/blob/main/packages/btp-utils/src/destination.ts#L21)

Reply:
@cianmSAP as discussed also with BAS colleagues, property `webIDEEnabled` can be completely removed. While it needs to be set to true in the destination config, we would never see a destination that has not set `webIDEEnabled `.

Reply:
Do we still want to display the WebIDEEnabled column (all defaulted to true) in the results table ?

Reply:
I've removed WebIDEEnabled